### PR TITLE
drivers: can: allow calling can_set_bitrate() from userspace

### PR DESCRIPTION
**Pull request by henrikbrixandersen**
- Add syscall to allow calling can_set_bitrate() from userspace.
- Add test for can_set_bitrate() syscall.

### CODE REVIEW

**Review by karstenkoenig — DISMISSED** [review]

**Comment by henrikbrixandersen** [comment]
Rebased after 53ef68d4598b2f9005c5da3fc0b860ca1999d350.

File: tests/drivers/can/api/src/main.c [hunk 1]
```diff
@@ -634,6 +639,17 @@ static void test_set_bitrate_too_high(void)
...
 }
 
+/**
+ * @brief Test setting bitrate.
+ */
+static void test_set_bitrate(void)
+{
+	int err;
+
+	err = can_set_bitrate(can_dev, TEST_BITRATE, 0);
```
Comment (karstenkoenig) [inline tests/drivers/can/api/src/main.c:649]: Will this not fail for CONFIG_CAN_FD_MODE=y?

Comment (henrikbrixandersen) [reply tests/drivers/can/api/src/main.c:649]: It would, but CAN-FD is explicitly disabled for this test suite.

**Review by karstenkoenig — DISMISSED** [review]

**Comment by henrikbrixandersen** [comment]
Rebased to fix conflict after fba27b4bde47e97175adf87c8bcd06aadcd8a43b.

**Review by alexanderwachter — APPROVED** [review]

**Comment by henrikbrixandersen** [comment]
@karstenkoenig Please review and reapprove after the rebase.

**Review by karstenkoenig — APPROVED** [review]

**Pull request merged**